api/src/main/java/io/strimzi/api/kafka/model/kafka/cruisecontrol/HashLoginServiceApiUsers.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.kafka.cruisecontrol;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.common.Constants;
+import io.strimzi.api.kafka.model.common.PasswordSource;
+import io.strimzi.api.kafka.model.common.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Cruise Control's API users config
+ */
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"type", "valueFrom"})
+@EqualsAndHashCode
+public class HashLoginServiceApiUsers implements UnknownPropertyPreserving {
```

Comment:
The question is whether we need that now or can just change it later when we actually add another type 🤷

Reply:
It has an impact on other parts (see my other comments). I think that if we talk about being prepared for future implementations (by using different types) then we should be ready for that already. Otherwise, let's remove the field and say we support only the HashLogin format. I mean, do we really see new types for the future? Having freedom in the API but not a corresponding reality in the code, it's something I don't like. Of course, it's just my opinion.

Reply:
Well, your review, your call. But the API is the part you cannot change later. But you can change the Java code.

Reply:
Did some refactoring for this, let me know if it looks ok!